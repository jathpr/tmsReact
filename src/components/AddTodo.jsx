import React, { useState } from 'react'

export const AddTodo = ({ addTodo }) => {
	const [input, setInput] = useState('')

	const handleAdd = () => {
		addTodo({ text: input, done: false })
		setInput('')
	}

	return (
		<>
			<input value={input} onChange={(e) => setInput(e.target.value)} />
			<button onClick={handleAdd}>Add</button>
		</>
	)
}
