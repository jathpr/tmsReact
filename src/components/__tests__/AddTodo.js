import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import { AddTodo } from '../AddTodo'
import renderer from 'react-test-renderer'

test('shows AddTodo', () => {
	const testMessage = 'Add'
	render(<AddTodo />)

	// query* functions will return the element or null if it cannot be found
	// get* functions will return the element or throw an error if it cannot be found
	expect(screen.queryByText(testMessage)).not.toBeNull()

	fireEvent.change(screen.getByDisplayValue(''), {
		target: { value: 'chuck' },
	})

	expect(screen.getByDisplayValue('chuck')).toBeDefined()
})

it('AddTodo renders correctly', () => {
	const tree = renderer.create(<AddTodo />).toJSON()
	expect(tree).toMatchSnapshot()
})

it('AddTodo renders correctly', () => {
	const tree = renderer.create(<AddTodo />).toJSON()
	expect(tree).toMatchInlineSnapshot(`
		Array [
		  <input
		    onChange={[Function]}
		    value=""
		  />,
		  <button
		    onClick={[Function]}
		  >
		    Add
		  </button>,
		]
	`)
})
