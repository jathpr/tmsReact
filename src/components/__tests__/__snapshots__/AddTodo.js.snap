// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo renders correctly 1`] = `
Array [
  <input
    onChange={[Function]}
    value=""
  />,
  <button
    onClick={[Function]}
  >
    Add
  </button>,
]
`;
